kernel/aster-nix/src/net/socket/ip/datagram/mod.rs

```diff
@@ -231,6 +237,32 @@ impl FileLike for DatagramSocket {
         Some(self)
     }
 
+    fn ioctl(&self, cmd: IoctlCmd, arg: usize) -> Result<i32> {
+        match cmd {
+            IoctlCmd::FIONBIO => {
+                let is_nonblocking = read_val_from_user::<i32>(arg)? != 0;
+                let mut flags = self.status_flags();
+                flags.set(StatusFlags::O_NONBLOCK, is_nonblocking);
+                self.set_status_flags(flags)?;
+                Ok(0)
+            }
+            IoctlCmd::FIOASYNC => {
+                let is_async = read_val_from_user::<i32>(arg)? != 0;
+                let mut flags = self.status_flags();
+                // Did FASYNC state change ?
+                if flags.contains(StatusFlags::O_ASYNC).bitxor(is_async) {
```

Comment:
The `O_ASYNC` flag does not appear to be fully supported. For example, in status_flags, only `O_NONBLOCK` is supported. So, in reality, this is just a dummy implementation. I suggest adding a TODO to support `O_ASYNC`.